src/shared/bootspec.c

```diff
@@ -1455,52 +1459,78 @@ static void print_addon(
                 const char *addon_str) {
 
         printf("  %s: %s\n", addon_str, addon->location);
-        printf("      cmdline: %s%s\n", special_glyph(SPECIAL_GLYPH_TREE_RIGHT), addon->cmdline);
+        printf("      options: %s%s\n", special_glyph(SPECIAL_GLYPH_TREE_RIGHT), addon->cmdline);
+}
+
+static int indent_embedded_newlines(char *cmdline, char **ret_cmdline) {
+        _cleanup_free_ char *t = NULL;
+        _cleanup_strv_free_ char **ts = NULL;
+
+        assert(ret_cmdline);
+
+        if (!cmdline)
+                return 0;
```

Comment:
so we have this rule in systemd, that functions that fail should never touch their ret params, and functions that succeed should always initialize them. `ret_cmdline` is a return parameter, and in this shortcut here, you return success (rightfully), hence please make sure to initialize ret_cmdline. Or in other words, add `*ret_cmdline = NULL` or so here in the shortcut return path

Reply:
i.e. error return is typically if you return `< 0`, and sucess return is if you return `>= 0

Reply:
I didn't get this thing with the error: if there is an error in `indent_embedded_newlines` (like oom), what should I return? -1? And what should the caller of `indent_embedded_newlines` return if there is an error? `log_oom`?

Reply:
Ok so as I understand this, apart from removing the if (!cmdline), I will return -1 at each failed string manipulation in the same function, and then in the caller log_oom is return value of `indent_embedded_newlines` is < 0.